spray-json/shared/src/main/scala/spray/json/JsonParser.scala

```diff
@@ -165,51 +166,146 @@ class JsonParser(input: ParserInput, settings: JsonParserSettings = JsonParserSe
   private def DIGIT(): Boolean = cursorChar >= '0' && cursorChar <= '9' && advance()
 
   // http://tools.ietf.org/html/rfc4627#section-2.5
-  private def `string`(): Unit = {
-    if (cursorChar == '"') cursorChar = input.nextUtf8Char() else fail("'\"'")
-    sb.setLength(0)
-    while (`char`()) cursorChar = input.nextUtf8Char()
-    require('"')
+  private def `string`(): String = {
+    if (cursorChar != '"') fail("'\"'")
+    val res = stringFast(input.cursor)
     ws()
+    res
   }
 
-  private def `char`() =
-    // simple bloom-filter that quick-matches the most frequent case of characters that are ok to append
-    // (it doesn't match control chars, EOI, '"', '?', '\', 'b' and certain higher, non-ASCII chars)
-    if (((1L << cursorChar) & ((31 - cursorChar) >> 31) & 0x7ffffffbefffffffL) != 0L) appendSB(cursorChar)
-    else cursorChar match {
-      case '"' | EOI => false
-      case '\\' =>
-        advance(); `escaped`()
-      case c => (c >= ' ') && appendSB(c)
+  private var charBuffer = new Array[Char](settings.initialMaxStringCharacters)
+  private def stringFast(start: Int): String = {
+    /* Scan to the end of a simple string as fast as possible, no escapes, no unicode */
+    // so far unclear results if that is really helpful
+    /*@tailrec def parseOneFast(cursor: Int, charsRead: Int): String = {
+      val b = input.byteAt(cursor)
+      charBuffer(charsRead) = (b & 0xff).toChar
+      //println(s"At $cursor $charsRead read '${charBuffer(charsRead)}'")
+
+      if (b == '"') {
+        input.setCursor(cursor) // now at end '"'
+        advance()
+        new String(charBuffer, 0, charsRead)
+      } else if (((b - 32) ^ 60) <= 0) // trick from jsoniter-scala
+        parseOneSlow(cursor, charsRead) // fall back to slower parsing
+      else
+        parseOneFast(cursor + 1, charsRead + 1)
+    }*/
+
+    /*
+     * Do full string parsing. This subsumes the functionality from parseOneFast, however, just using
+     * this one for all parsing is significantly slower (> 10 %). A possible reason might be that using
+     * this function for all string parsing has worse branch prediction than the simple and fast variant?
+     */
+    @tailrec def parseOneSlow(cursor: Int, charsRead: Int): String = {
+      val b = input.charAt(cursor)
+      charBuffer(charsRead) = b
+
+      if (b == '"') {
+        input.setCursor(cursor) // now at end '"'
+        advance()
+        new String(charBuffer, 0, charsRead)
+      } else if (b == '\\') {
+        val skip = escaped(cursor, charsRead)
+        parseOneSlow(cursor + skip, charsRead + 1)
+      } else if (input.needsDecoding && b >= 128) {
```

Comment:
missing check for control characters that [must](https://tools.ietf.org/html/rfc8259#section-7) be escaped